packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -2035,9 +2035,14 @@ class StackViewport extends Viewport implements IStackViewport {
 
     // Create a VTK Image Slice actor to display the vtkImageData object
     const actor = this.createActorMapper(this._imageData);
-    const actors = [];
-    actors.push({ uid: this.id, actor });
-    this.setActors(actors);
+    const oldActors = this.getActors();
+    if (oldActors.length && oldActors[0].uid === this.id) {
+      oldActors[0].actor = actor;
+    } else {
+      oldActors.unshift({ uid: this.id, actor });
+    }
+    this.setActors(oldActors);
```

Comment:
can you explain what is happening here?

Reply:
If one adds an actor in the stackViewport this part of code will remove it from the actors list, because old code creates a list with only the ImageSlice actor and sets it as the actors list. So what I am doing here is checking if the imageslice actor is in the actor list. If not adding as the first item of the list

Reply:
I'm not following unfortunately. 

So this says that if the uid of the actor is the same as the viewportId (what does that even mean), then replace the actor at the first index? otherwise throw it out?

Why?